utils/src/main/java/io/questdb/cliutil/TxFileStruct.java

```diff
@@ -46,19 +45,56 @@ class TxFileStruct {
     public long TX_OFFSET_TRUNCATE_VERSION;
     public long TX_OFFSET_TXN;
 
-    public long calculateFileSize() {
-        return getPartitionTableIndexOffset(TX_OFFSET_MAP_WRITER_COUNT, ATTACHED_PARTITION_SIZE * 4) + TX_BASE_HEADER_SIZE;
+    @Override
+    public String toString() {
+        return "TxFileStruct{" +
+                "TX_OFFSET_TXN=" + TX_OFFSET_TXN +
+                ", TX_OFFSET_COLUMN_VERSION=" + TX_OFFSET_COLUMN_VERSION +
+                ", TX_OFFSET_FIXED_ROW_COUNT=" + TX_OFFSET_FIXED_ROW_COUNT +
+                ", TX_OFFSET_TRANSIENT_ROW_COUNT=" + TX_OFFSET_TRANSIENT_ROW_COUNT +
+                ", TX_OFFSET_STRUCT_VERSION=" + TX_OFFSET_STRUCT_VERSION +
+                ", TX_OFFSET_DATA_VERSION=" + TX_OFFSET_DATA_VERSION +
+                ", TX_OFFSET_PARTITION_TABLE_VERSION=" + TX_OFFSET_PARTITION_TABLE_VERSION +
+                ", TX_OFFSET_TRUNCATE_VERSION=" + TX_OFFSET_TRUNCATE_VERSION +
+                ", TX_OFFSET_MAP_WRITER_COUNT=" + TX_OFFSET_MAP_WRITER_COUNT +
+                ", SYMBOLS=" + SYMBOLS +
+                ", TX_OFFSET_MAX_TIMESTAMP=" + TX_OFFSET_MAX_TIMESTAMP +
+                ", TX_OFFSET_MIN_TIMESTAMP=" + TX_OFFSET_MIN_TIMESTAMP +
+                ", ATTACHED_PARTITIONS=" + ATTACHED_PARTITIONS +
+                ", ATTACHED_PARTITIONS_COUNT=" + ATTACHED_PARTITIONS_COUNT +
+                ", TX_OFFSET_SEQ_TXN=" + TX_OFFSET_SEQ_TXN +
+                '}';
     }
 
     static class AttachedPartition {
+        long AVAILABLE0;
+        long AVAILABLE1;
+        long AVAILABLE2;
         long DATA_TX;
+        long MASK;
+        long MASKED_SIZE;
```

Comment:
Can we have SIZE and RO as separate fields?

Reply:
mmmm yeah, I guess that would be better.